### DISCUSSION TITLE
Payload examples for OAS

### DISCUSSION THREAD

**Discussion by BernieWhite** [microsoft/typespec#2047] [Q&A] [answered]
Just wanting to know if there is an easy way to generate payload examples for OAS 3 specs.

I was hoping there was an easy `@example` option I could use to specify an example model for request/ response payload schemas.

Any suggestions of where to start?

**Comment by NielsCo** [comment, marked as answer]
I also didn't find any easy way to do this. Maybe write an Issue for this?

I have a suggestion for a _very wonky_ workaround.
You could use an extension of OAS and then remove the "x" in a script.

Your example would be defined like this:
```
@extension("x-example", {id: "test", weight: 23123, color: "blue"})
model Widget {
  @key id: string;
  weight: int32;
  color: "red" | "blue";
}
```
Here is a [playground-link](https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZb3BlbmFwadwcMyI7CkBzZXJ2aWNlKHsKICB0aXRsZTogIldpZGdldCBTxhsiLAp9KQpuYW1lc3BhY2UgRGVtb8cbOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TIFU9wZW5BUEk7CgoKQGV4dGVuc2lvbigieC1leGFtcGxlIiwge2lkOiAidOQA2iwgd2VpZ2h0OiAyMzEyMywgY29sb3I6ICJibHVlIn0pCm1vZGVsIOcAseQAxEBrZXkgxEFzdHJpbmc7CiDJQ2ludDMyxBHIRXJlZCIgfMdNOwp9CgpvcCBnZXQoQHBhdGgg%2BACgIjAiKctjKSA65wCBOw%3D%3D)
And then you would need to write a script that removes the "x-" 
Obviously this isn't type-checked or anything as you can define anything you'd like to be an extension.

Type-Checked examples would be a very cool feature for TypeSpec IMHO.
If you're too busy to write the issue, let me know and I'll write it. Didn't want to take credit for it.

**Reply by BernieWhite** [reply]
Logged as https://github.com/microsoft/typespec/issues/2087
